test/controllers/gobierto_data/api/v1/datasets_deletion_test.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module GobiertoData
+  module Api
+    module V1
+      class DatasetsDeletionControllerTest < GobiertoControllerTest
+        self.use_transactional_tests = false
+
+        def auth_header
+          @auth_header ||= "Bearer #{admin.primary_api_token}"
+        end
+
+        def site
+          @site ||= sites(:madrid)
+        end
+
+        def admin
+          @admin ||= gobierto_admin_admins(:tony)
+        end
+
+        # DELETE /api/v1/data/datasets/:dataset-slug
+        def test_delete_dataset_remove_also_visualizations_favourites_queries_gobierto_data_table
+          with(site: site) do
+            dataset = gobierto_data_datasets(:dataset_to_delete)
+            query = gobierto_data_queries(:dataset_to_delete_query_to_delete)
+            visualization = gobierto_data_visualizations(:dataset_to_delete_visualization_to_delete)
+            gdata_db_name = gobierto_module_settings(:gobierto_data_settings_madrid).settings["db_config"]["read_db_config"]["database"]
+
```

Comment:
Could you add here this assert?

`assert ::GobiertoData::Connection.execute_query(site, "select exists (select from information_schema.tables where table_schema = '#{gdata_db_name}' and table_name = '#{dataset.table_name}')").first["exists"]`

and run tests again? I'm afraid it always returns false

Reply:
Failure:
GobiertoData::Api::V1::DatasetsDeletionControllerTest#test_delete_dataset_remove_also_visualizations_favourites_queries_gobierto_data_table [/home/stbn/populate-tools/gobierto/test/controllers/gobierto_data/api/v1/datasets_deletion_test.rb:30]
Minitest::Assertion: Expected false to be truthy

after to commit this a try value putting a byebug 
(byebug) ::GobiertoData::Connection.execute_query(site, "select exists (select from information_schema.tables where table_schema = '#{gdata_db_name}' and table_name = '#{dataset.table_name}')").first["exists"]
false

you are right better if rewrite as:
```
assert_equal false, ::GobiertoData::Connection.execute_query(site, "select exists (select from information_schema.tables where table_schema = '#{gdata_db_name}' and table_name = '#{dataset.table_name}')").first["exists"]
```